www/js/util.js

```diff
@@ -688,10 +690,42 @@ function saveUserOptions() {
         USEROPTS.show_shadowchat = $("#us-shadowchat").prop("checked");
     }
 
+    getNotificationPermission();
+
     storeOpts();
     applyOpts();
 }
 
+function getNotificationPermission() {
+    //Only ask for permission if they desire the permission.
+    if(USEROPTS.desktop_notification != "never")
```

Comment:
Braces on the same line

```js
if (...) {
```

Use `!==` rather than `!=` for consistency (even though it won't cause any type coercion issues in this specific scenario)